src/top-bar/period-select/period-select.js

```diff
@@ -29,7 +29,12 @@ const PeriodSelect = () => {
 
     useEffect(() => {
         if (workflow?.periodType) {
-            setMaxYear(computeMaxYear(workflow?.periodType))
+            const newMaxYear = computeMaxYear(workflow?.periodType)
+            setMaxYear(newMaxYear)
+
+            if (!period?.year) {
+                setYear(newMaxYear)
+            }
```

Comment:
I would say this part is the actual fix. If `workflow.periodType` is not available on first mount the `year` and `maxYear` are both `null`. This is fine, but as soon as the periodType does become available the year needs to be updated too.

I only do this when `period.year` is unavailable, since the presence of `period.year` indicates a user-selection.